### PR TITLE
[various] Speed up build-examples

### PR DESCRIPTION
**Pull request by reidbaker**
https://github.com/flutter/flutter/issues/125707

- Create pluginTools config for every example that shrunk material icons and had an android config

Likely some build caching but before/after If someone knows how to clean the cache or force a rebuild I will give more accurate after numbers. 
Command
`time dart run ./script/tool/bin/flutter_plugin_tools.dart build-examples --apk`
Secondary validation from @stuartmorgan who discovered this is between a 10% and 25% speed increase. 

Existing tests add confidence this is non-breaking. New tests not required because these are optional flags. Validation this does not change is low consequence.  Test update to exclude these config files from changelog requirements. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/animations/example/android/.pluginToolsConfig.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+buildFlags:
+  _pluginToolsConfigGlobalKey:
+    - "--no-tree-shake-icons"
```
Comment (Piinks) [inline packages/animations/example/android/.pluginToolsConfig.yaml:3]: Do all of these need EOF newlines? Or is github complaining unnecessarily? :)

Comment (reidbaker) [reply packages/animations/example/android/.pluginToolsConfig.yaml:3]: They all needed new lines. Updated.

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

**Review by chunhtai — APPROVED** [review]
LGTM

**Comment by reidbaker** [comment]
Test exemption requested https://discord.com/channels/608014603317936148/608021234516754444/1102644603594739722

**Comment by Hixie** [comment]
test exemption should not be necessary, the PR modifies a test

File: script/tool/lib/src/common/package_state_utils.dart [hunk 2]
```diff
@@ -181,6 +181,8 @@ Future<bool> _isDevChange(List<String> pathComponents,
       pathComponents.contains('lint-baseline.xml') ||
       // Example build files are very unlikely to be interesting to clients.
       _isExampleBuildFile(pathComponents) ||
+      // Build flags are very unlikely to be interesting to clients.
+      pathComponents.contains('.pluginToolsConfig.yaml') ||
```
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply script/tool/lib/src/common/package_state_utils.dart:185]: Could you move this into `_isExampleBuildFile`? I wouldn't expect this to be outside of `example`.

**Review by stuartmorgan — APPROVED** [review]
LGTM with nit.

**Comment by reidbaker** [comment]
> test exemption should not be necessary, the PR modifies a test

Sorry test modification added after the request was sent.

File: script/tool/lib/src/common/package_state_utils.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (reidbaker) [reply script/tool/lib/src/common/package_state_utils.dart]: done

**Pull request merged**